primitives/src/traits.rs

```diff
@@ -34,7 +34,7 @@ pub struct AMMTransfer<AccountId, AssetId, AssetPair, Balance> {
 
 /// Traits for handling AMM Pool trades.
 pub trait AMM<AccountId, AssetId, AssetPair, Amount> {
-	/// Check if both assets exist in a pool.
+	/// Check if both assets exist in a pool. //TODO: not sure if AssetPair is right here since we don't have asset_in or asset_out when not trading.
```

Comment:
Yeah, i was actually thinking that this AMM trait should be more generic to cover various pool implementations.

I have some ideas - but do we want to refactor now ?

Reply:
Probably not. But let's start working on it?

Reply:
After launch of crowdloan

Reply:
#145